src/Components/Components/src/BindMethods.cs

```diff
@@ -69,134 +69,200 @@ public static MulticastDelegate GetEventHandlerValue<T>(Func<T, Task> value)
             return value;
         }
 
+        /// <summary>
+        /// Not intended to be used directly.
+        /// </summary>
+        public static EventCallback GetEventHandlerValue<T>(EventCallback value)
+            where T : UIEventArgs
+        {
+            return value;
+        }
+
+        /// <summary>
+        /// Not intended to be used directly.
+        /// </summary>
+        public static EventCallback<T> GetEventHandlerValue<T>(EventCallback<T> value)
+            where T : UIEventArgs
+        {
+            return value;
+        }
+
         /// <summary>
         /// Not intended to be used directly.
         /// </summary>
         public static Action<UIEventArgs> SetValueHandler(Action<string> setter, string existingValue)
         {
-            return _ => setter((string)((UIChangeEventArgs)_).Value);
+            return eventArgs =>
+            {
+                setter((string)((UIChangeEventArgs)eventArgs).Value);
+                _ = DispatchEventAsync(setter.Target, EventCallbackWorkItem.Empty, UIEventArgs.Empty);
+            };
         }
 
         /// <summary>
         /// Not intended to be used directly.
         /// </summary>
         public static Action<UIEventArgs> SetValueHandler(Action<bool> setter, bool existingValue)
         {
-            return _ => setter((bool)((UIChangeEventArgs)_).Value);
+            return eventArgs =>
+            {
+                setter((bool)((UIChangeEventArgs)eventArgs).Value);
+                _ = DispatchEventAsync(setter.Target, EventCallbackWorkItem.Empty, UIEventArgs.Empty);
```

Comment:
Explained this in a comment elsewhere but for visibility. Yes, this isn't *super* correct. It's a temporary polyfill until we swap over to the new infrastructure. This is needed before of the change to event dispatching in the renderer. If we don't do something like this we're going to *chicken-or-egg* super hard 🐔

Reply:
I haven't completely followed in what sense this isn't correct. Will try to see if the comment elsewhere enlightens me.

Reply:
What is incorrect about is the error handling. It calls `DispatchEventAsync` as fire-and-forget.